### PR TITLE
Fix conflict endpoint URLs

### LINKED ISSUE #334: Conflicts endpoint returns incorrect URL's
At the moment, the conflicts endpoint is serializing projects and students like this:

```kotlin
val baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString()
// ...
studentsMap[student.id]!!.add("$baseUrl/projects/" + project.id)
// ...
conflicts.add(Conflict("$baseUrl/students/$studentId", projectIds))
```

The baseUrl doesn't contain an `{edition}`, which means the serialized URL doesn't contain one either and is therefore pointing to an incorrect location. The URL's should be fixed to include the `{edition}`. It would be best to do this by using URL serializers (see [here](https://github.com/SELab-2/OSOC-1/blob/development/backend/src/main/kotlin/be/osoc/team1/backend/util/Serializers.kt)), as opposed to keeping it hardcoded like this.

### PR DESCRIPTION
**Pull request by TomAlard**
Closes #334.

**Breaking change checklist**:
- [x] I've updated the existing tests to account for the changes in expected behavior
- [x] I've updated the `osoc.yaml` file

### CODE REVIEW

File: backend/src/main/kotlin/be/osoc/team1/backend/services/ProjectService.kt [hunk 1]
```diff
@@ -98,20 +99,21 @@ class ProjectService(
...
-                studentsMap.putIfAbsent(student.id, mutableListOf())
-                studentsMap[student.id]!!.add("$baseUrl/projects/" + project.id)
+                studentsMap.putIfAbsent(student, mutableListOf())
+                studentsMap[student]!!.add(projectSerializer.toUrl(project))
             }
         }
         val conflicts = mutableListOf<Conflict>()
-        for ((studentId, projectIds) in studentsMap.entries) {
+        val studentSerializer = StudentSerializer()
+        for ((student, projectIds) in studentsMap.entries) {
```
Comment (Mikxox) [inline backend/src/main/kotlin/be/osoc/team1/backend/services/ProjectService.kt:113]: Maybe change 'projectIds' to 'projectUrls' or something else more descriptive.

**Review by Mikxox — APPROVED** [review]

File: backend/src/main/kotlin/be/osoc/team1/backend/services/ProjectService.kt (hunk 1 above)
Comment (NinjaJay) [inline backend/src/main/kotlin/be/osoc/team1/backend/services/ProjectService.kt:113]: I know this isn't your fault, but could you change projectIds to ProjectURLs? That way the parameter is more descriptive.

**Review by NinjaJay — APPROVED** [review]
It's good after you change the comment of Michael.

File: backend/src/main/kotlin/be/osoc/team1/backend/services/ProjectService.kt (hunk 1 above)
Comment (TomAlard) [reply backend/src/main/kotlin/be/osoc/team1/backend/services/ProjectService.kt]: Done

**Pull request merged**